### PR TITLE
Commander: Preflight check improvement

### PR DESCRIPTION
**Pull request by Zefz**
Improves the user feedback for preflight checks. If a sensor fails because it is not calibrated or healthy, an alarm tune will play *instead* of the normal startup tune. Furthermore the LED will blink rapidly red until the problem is resolved and the system will refuse to arm (with appropriate error messages).

This moves the preflight check to the commander module and effectively deprecates the old ```preflight_check``` module (I would recommend removing it to avoid having to maintain two systems which do the same).

Please do not merge into master until I have had time to verify my fixes. One potential problem is in-air reboot where the system will run pre-flight checks and fail (because of accelerations, 1.5 second delay while collecting sensor data and what-not)

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Awesome! I will try to test the inflight reboot - as the preflight check is already disabled it shouldn't actually lock.

**Comment by Zefz** [comment]
@LorenzMeier This is the line I'm talking about:
https://github.com/Zefz/Firmware/commit/a20d8ece0742f6013324651b9b21f9529824958c#diff-a2bcb238e25d8a97ab2aee48f54dbc48R182
which was part of ```preflight_check.c```. The ```preflight_check.c``` module ran in a new thread so it didn't block the boot logic, but the commander runs this in it's main task thread.

**Comment by LorenzMeier** [comment]
I would suggest removing that line, I don't think it's strictly needed. In addition, the state restore logic should exclude the execution of the check completely, for this and other reasons. I did test a while back that it does, but I would indeed need to re-test.

**Comment by Zefz** [comment]
Is there a way the Commander knows that it is doing an in-air reboot in it's startup routines? (some sort of flag set?)

**Comment by LorenzMeier** [comment]
@Zefz Here is the code path for in-air reboot:
https://github.com/PX4/Firmware/blob/master/src/modules/commander/commander.cpp#L549-L554

This command is sent by the IO driver, which in turns uses the IO onboard state for state preservation. This will be moved to the battery-backed SRAM once available, but the command will likely remain similar.

You can see that the pre-arm checks are completely bypassed on a restore:
https://github.com/PX4/Firmware/blob/master/src/modules/commander/state_machine_helper.cpp#L153-L155

**Comment by Zefz** [comment]
That looks good. There was some unclear code (possibly legacy?) that indicated that the entire commander could *reboot* in-air (in air reboot as opposed to in air arming restart). It could be that I am still confusing terminologies/functionality here. There is also a difference between pre-flight check and pre-arm check ;)

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -536,9 +537,16 @@ bool handle_command(struct vehicle_status_s *status_local, const struct safety_s
 		break;
 
 	case VEHICLE_CMD_COMPONENT_ARM_DISARM: {
+
+			//Refuse to arm if preflight checks have failed
```
Comment (LorenzMeier) [inline src/modules/commander/commander.cpp]: This check needs to be moved to this location a few lines below:
https://github.com/PX4/Firmware/pull/1895/files#diff-1d51a813dcb26f46616f71f28aacc70dL543

See the comment about in-air restore. If this is an in-air restore event, the checks should not be executed.

**Comment by LorenzMeier** [comment]
Much appreciated! I will try to get some testing done on this the coming week. If not I might have to rely on you to do the last change I suggested and testing with @kd0aij.

**Comment by LorenzMeier** [comment]
From the commits and diff this looks like its good to go - is this accurate?

**Comment by Zefz** [comment]
@LorenzMeier  Yes. It might need rebasing on master again, it has been a while since it was synced.

**Comment by LorenzMeier** [comment]
Closing and reopening a rebased one.

**Pull request closed without merging**